src/main/webapp/app/exercises/shared/structured-grading-criterion/grading-instructions-details/grading-instructions-details.component.ts

```diff
@@ -285,4 +320,113 @@ export class GradingInstructionsDetailsComponent implements OnInit {
         this.exercise.gradingCriteria = [];
         this.createSubInstructionCommands(domainCommands);
     }
+
+    /**
+     * @function onInstructionChange
+     * @desc 1. Gets a tuple of text and domainCommandIdentifiers and assigns text values according to the domainCommandIdentifiers
+     *       2. The tupple order is the same as the order of the commands in the markdown text inserted by the user
+     * @param domainCommands containing tuples of [text, domainCommandIdentifiers]
+     * @param {GradingInstruction} instruction
+     * @param {GradingCriterion} criteria
+     */
+    onInstructionChange(domainCommands: [string, DomainCommand | null][], instruction: GradingInstruction): void {
+        this.instructions = [];
+        this.instructions.push(instruction);
+        this.setInstructionParameters(domainCommands);
+    }
+
+    /**
+     * @function resetInstruction
+     * @desc Resets the whole instruction
+     * @param instruction {GradingInstruction} the instruction, which will be reset
+     * @param criteria {GradingCriterion} the criteria, which includes the instruction that will be reset
+     */
+    resetInstruction(instruction: GradingInstruction, criteria: GradingCriterion) {
+        const criteriaIndex = this.exercise.gradingCriteria!.findIndex((gradingCriteria) => {
+            return gradingCriteria.id === criteria.id;
+        });
+        const instructionIndex = this.exercise.gradingCriteria![criteriaIndex].structuredGradingInstructions.findIndex((sgi) => {
+            return sgi.id === instruction.id;
+        });
+
+        const backupCriteriaIndex = this.backupExercise.gradingCriteria!.findIndex((gradingCriteria) => {
+            return gradingCriteria.id === criteria.id;
+        });
+        const backupInstructionIndex = this.backupExercise.gradingCriteria![criteriaIndex].structuredGradingInstructions.findIndex((sgi) => {
```

Comment:
Shouldn't this be [backupCriteriaIndex]? I don't know the code but I am asking to be sure